src/promptflow/promptflow/executor/_process_manager.py

```diff
@@ -0,0 +1,327 @@
+            flow_file,
+            connections,
+            working_dir,
+            *args,
+            **kwargs):
+        super().__init__(*args, **kwargs)
+        self._control_signal_queue = control_signal_queue
+        self._flow_file = flow_file
+        self._connections = connections
+        self._working_dir = working_dir
+
+    def start_processes(self):
+        '''
+        Initiates a process using the 'spawn' method.
+        '''
+        context = multiprocessing.get_context("spawn")
+        process = context.Process(
+            target=fork_processes_manager,
+            args=(
+                self._log_context_initialization_func,
+                self._current_operation_context,
+                self._input_queues,
+                self._output_queues,
+                self._control_signal_queue,
+                self._flow_file,
+                self._connections,
+                self._working_dir,
+                self._raise_ex,
+                self._process_info,
+                self._process_target_func
+            )
+        )
+        process.start()
+
+    def restart_processes(self, i):
+        """
+        Sends a signal to restart a specific process.
+
+        :param i: Index of the process to restart.
+        :type i: int
+        """
+        self._control_signal_queue.put(('restart', i))
+
+    def end_process(self, i):
+        """
+        Sends a signal to terminate a specific process.
+
+        :param i: Index of the process to terminate.
+        :type i: int
+        """
+        self._control_signal_queue.put(('end', i))
+
+    def new_process(self, i):
+        """
+        Sends a signal to start a new process.
+
+        :param i: Index of the new process to start.
+        :type i: int
+        """
+        self._control_signal_queue.put(('start', i))
+
+
+def fork_processes_manager(
+        log_context_initialization_func,
+        current_operation_context,
+        input_queues,
+        output_queues,
+        control_signal_queue,
+        flow_file,
+        connections,
+        working_dir,
+        raise_ex,
+        process_info,
+        process_target_func
+):
+    """
+    Manages the creation, termination, and signaling of processes using the 'fork' context.
+    """
+    # Set up signal handling for process interruption.
+    from promptflow.executor._line_execution_process_pool import signal_handler, create_executor_fork
+    signal.signal(signal.SIGINT, signal_handler)
+
+    # Set the multiprocessing context to 'fork'.
+    context = multiprocessing.get_context("fork")
+
+    # Create flow executor.
+    executor = FlowExecutor.create(
+        flow_file=flow_file,
+        connections=connections,
+        working_dir=working_dir,
+        raise_ex=raise_ex,
+    )
+
+    executor_creation_func = partial(create_executor_fork, flow_executor=executor)
+
+    # Function to create and start a new process.
+    def new_process(i):
```

Comment:
We might extract these logic in another new class `SpawnedForkProcessManager`, then the logic could be more clear like:
```
def fork_process_manager():
    manager = SpawnedForkProcessManager()
    for i in range(len(input_queues)):
        manager.new_process(i)
    while True:
        # handle signals
```

Reply:
fixed